packages/replay/src/upload.ts

```diff
@@ -8,278 +8,259 @@ import { Options, OriginalSourceEntry, RecordingMetadata, SourceMapEntry } from
 
 const debug = dbg("replay:cli:upload");
 
-let gClient: ProtocolClient | undefined;
-let gClientReady = defer<boolean>();
-
-async function initConnection(
-  server: string,
-  accessToken?: string,
-  verbose?: boolean,
-  agent?: any
-) {
-  if (!gClient) {
-    let { resolve } = gClientReady;
-    gClient = new ProtocolClient(
-      server,
-      {
-        async onOpen() {
-          try {
-            await gClient!.setAccessToken(accessToken);
-            resolve(true);
-          } catch (err) {
-            maybeLog(verbose, `Error authenticating with server: ${err}`);
+// Granularity for splitting up a recording into chunks for uploading.
+const ChunkGranularity = 1024 * 1024;
+
+function sha256(text: string) {
+  return crypto.createHash("sha256").update(text).digest("hex");
+}
+
+class ReplayClient {
+  client: ProtocolClient | undefined;
+  clientReady = defer<boolean>();
+
+  async initConnection(server: string, accessToken?: string, verbose?: boolean, agent?: any) {
```

Comment:
small nit, but does this need to be async?

Reply:
No. It's returning a promise either way but the `async` isn't required since we're not `await`-ing anything.